From 8de824a15fe9d97fc169b262c884e09f6e59912a Mon Sep 17 00:00:00 2001
From: Jeremy Kerr <jk@ozlabs.org>
Date: Wed, 8 Jul 2015 15:59:40 +0800
Subject: [PATCH 3/3] console/ast2400: Fix swapped address/value writes in
 writeSIOReg

We need to write the address to 0x2e, and the data to 0x2f.

Change-Id: I1271a134c19d921fa1ee54b5f0d1c0e092df9e2b
Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
Reviewed-on: http://gfw160.aus.stglabs.ibm.com:8080/gerrit/19146
Tested-by: Jenkins Server
Reviewed-by: A. Patrick Williams III <iawillia@us.ibm.com>
---
 src/usr/console/ast2400.C | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/usr/console/ast2400.C b/src/usr/console/ast2400.C
index c463dcd..243c9be 100644
--- a/src/usr/console/ast2400.C
+++ b/src/usr/console/ast2400.C
@@ -83,11 +83,11 @@ namespace CONSOLE
 
             do{
 
-                l_err = _writeReg( SIO_ADDR_REG_2E, i_data );
+                l_err = _writeReg( SIO_ADDR_REG_2E, i_reg );
 
                 if(l_err) { break; }
 
-                l_err = _writeReg( SIO_DATA_REG_2F, i_reg );
+                l_err = _writeReg( SIO_DATA_REG_2F, i_data );
 
             }while(0);
 
-- 
2.4.4

